src/main/scala/org/ergoplatform/modifiers/history/BlockTransactions.scala

```diff
@@ -69,32 +66,23 @@ object BlockTransactions extends ApiCodecs {
   }
 }
 
-object BlockTransactionsSerializer extends Serializer[BlockTransactions] {
-  override def toBytes(obj: BlockTransactions): Array[Byte] = {
-    val txsBytes = concatBytes(obj.txs.map { tx =>
-      val txBytes = ErgoTransactionSerializer.toBytes(tx)
-      Bytes.concat(Ints.toByteArray(txBytes.length), txBytes)
-    })
-    Bytes.concat(idToBytes(obj.headerId), txsBytes)
-  }
+object BlockTransactionsSerializer extends ScorexSerializer[BlockTransactions] {
 
-  override def parseBytes(bytes: Array[Byte]): Try[BlockTransactions] = Try {
-    val headerId: ModifierId = bytesToId(bytes.slice(0, Constants.ModifierIdSize))
-
-    def parseTransactions(index: Int, acc: Seq[ErgoTransaction]): BlockTransactions = {
-      if (index == bytes.length) {
-        BlockTransactions(headerId, acc)
-      } else {
-        val txLength = Ints.fromByteArray(bytes.slice(index, index + 4))
-        require(txLength > 0)
-        val tx = ErgoTransactionSerializer.parseBytes(bytes.slice(index + 4, index + 4 + txLength)) match {
-          case Success(parsedTx) => parsedTx
-          case Failure(f) => throw f
-        }
-        parseTransactions(index + 4 + txLength, acc :+ tx)
-      }
+  override def serialize(obj: BlockTransactions, w: Writer): Unit = {
+    w.putBytes(idToBytes(obj.headerId))
+    w.putInt(obj.txs.size)
```

Comment:
VLQ writer should be better here, for now number of transactions in block is small